pkg/client/dtclient/settings_client.go

```diff
@@ -608,18 +610,59 @@ func (d *SettingsClient) Upsert(ctx context.Context, obj SettingsObject, upsertO
 	resp, err := SendWithRetryWithInitialTry(ctx, d.client.POST, d.settingsObjectAPIPath, corerest.RequestOptions{CustomShouldRetryFunc: corerest.RetryIfTooManyRequests}, payload, retrySetting)
 	if err != nil {
 		d.settingsCache.Delete(obj.SchemaId)
-		return DynatraceEntity{}, fmt.Errorf("failed to create or update Settings object with externalId %s: %w", externalID, err)
+		return DynatraceEntity{}, fmt.Errorf("failed to create or update settings object with externalId %s: %w", externalID, err)
 	}
 
 	entity, err := parsePostResponse(resp.Data)
 	if err != nil {
 		return DynatraceEntity{}, err
 	}
 
+	if featureflags.AccessControlSettings.Enabled() && upsertOptions.AllUserPermission != nil {
+		permErr := d.modifyPermission(ctx, entity.Id, *upsertOptions.AllUserPermission)
+
+		if permErr != nil {
+			return DynatraceEntity{}, fmt.Errorf("failed to modify permissions of settings object with externalId %s: %w", externalID, permErr)
+		}
+	}
+
 	log.WithCtxFields(ctx).Debug("Created/Updated object %s (%s) with externalId %s", obj.Coordinate.ConfigId, obj.SchemaId, externalID)
 	return entity, nil
 }
 
+// modifyPermission creates, updates or deletes the all-user permission of a given settings object
+func (d *SettingsClient) modifyPermission(ctx context.Context, objectID string, allUserPermission config.AllUserPermissionKind) error {
+	permissions := getPermissionsFromConfig(allUserPermission)
+
+	if permissions == nil {
+		err := d.DeletePermission(ctx, objectID)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := d.UpsertPermission(ctx, objectID, PermissionObject{
+			Permissions: permissions,
+			Accessor:    &Accessor{AllUsers},
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
```suggestion
func (d *SettingsClient) modifyPermission(ctx context.Context, objectID string, allUserPermission config.AllUserPermissionKind) error {
	permissions := getPermissionsFromConfig(allUserPermission)

	if permissions != nil {
		return d.UpsertPermission(ctx, objectID, PermissionObject{
			Permissions: permissions,
			Accessor:    &Accessor{AllUsers},
		})
	}

	return d.DeletePermission(ctx, objectID)
}
```
I would simplify this as the method returns error and you not wrapping the error anyway. Also removed the else as I don't like it :). I ran the unit test and all passed, but please re-check.